Add `tolerations` to `Placement` for GRC

dhaiducek opened the pull request:
Policy and PolicySets should not respond to unreachable clusters, so we need to add tolerations to Placement by default. Tolerations didn't exist previously, so I've added them to `PlacementSection` through a `defaultPlacementSpec`.

In addition:
- Removes references to "ManagedClusterSetBindings" (since this is a kind the console user doesn't need to know about on the GRC page)
- Updates tooltip wording to be more accurate
- Surfaces a previously commented out "description" for Placement

ref: https://issues.redhat.com/browse/ACM-10200

dhaiducek commented:
/hold for reviews

/cc @mprahl @yiraeChristineKim

yiraeChristineKim commented on frontend/src/wizards/Placement/PlacementSection.tsx:
```diff
@@ -240,7 +248,7 @@ export function PlacementSection(props: {
               createClusterSetCallback={props.createClusterSetCallback}
               alertTitle={t(
-                'ClusterSets failed to load. Check the ManagedClusterSetBinding resource to verify your selected namespace.'
+                'ClusterSets failed to load. Verify that there are ClusterSets bound to your selected namespace.'
```
Should translate to diff langs?

yiraeChristineKim replied:
```
"i18n:fix:frontend": "cd frontend && npm run i18n:fix --",
```

yiraeChristineKim commented:
/approve

dhaiducek commented:
Please take another look. I've made some updates:
- Changed the `tolerations` property to `defaultPlacementSpec` to provide additional flexibility for the future
- Enabled and edited a previously commented-out "description" (to be reviewed also by the Doc team)
- Updated some tooltip wording (the ClusterSet tooltip had been copied from Label expressions so was inaccurate)
- Moved the ClusterSet alert up for visibility and in an attempt to decouple it from the ClusterSets field.

dockerymick commented on frontend/src/wizards/Placement/PlacementSection.tsx:
```diff
@@ -203,7 +210,7 @@ export function PlacementSection(props: {
   return (
     <Section
       label={t('Placement')}
-      // description="Placement selects clusters from the cluster sets which have bindings to the resource namespace."
+      description="Placement selects clusters from the cluster sets which have been bound to the resource namespace. An empty Placement returns all available clusters from all bound cluster sets."
```
Use Placement resources to select clusters from the cluster sets that you have bound to the resource namespace. An empty Placement returns all available clusters from all bound cluster sets.

dockerymick commented on frontend/public/locales/en/translation.json:
```diff
@@ -2303,9 +2302,9 @@
   "Select an inventory": "Select an inventory",
   "Select at least one cluster set to deploy application resources.": "Select at least one cluster set to deploy application resources.",
   "Select cluster label": "Select cluster label",
+  "Select cluster sets from which to select clusters. If no cluster sets are selected, all clusters will be selected from all cluster sets bound to the namespace.": "Select cluster sets from which to select clusters. If no cluster sets are selected, all clusters will be selected from all cluster sets bound to the namespace.",
```
```suggestion
  "Select cluster sets from which to select clusters. If you do not select a cluster set, all clusters are selected from all cluster sets bound to the namespace.": "Select cluster sets from which to select clusters. If you do not select a cluster set, all clusters are selected from all cluster sets bound to the namespace.",
```

dockerymick reviewed (changes requested):
left some suggestions

dhaiducek commented:
/unhold

dhaiducek commented:
Thanks for the reviews, all! I think all relevant parties have reviewed this PR, so it should be good to re-review and merge.

dhaiducek commented:
...and by "ready", I mean as soon as I fix the linting... 🤦‍♂️

dhaiducek commented:
Obligatory screenshots:
![image](https://github.com/stolostron/console/assets/19750917/9ba5da65-1066-4f2f-994b-a4ad4342fab6)
![image](https://github.com/stolostron/console/assets/19750917/9c24fae3-27c0-4032-a7f9-981d01100d35)

dhaiducek commented:
/cherry-pick release-2.10
